src/test/java/com/epam/tc/hw4/utils/AssertStep.java

```diff
@@ -0,0 +1,122 @@
+    }
+
+    @Step("Assert 4 images on the index page")
+    public void assertImages() {
+        indexPage = new IndexPage(driver);
+        for (WebElement i : indexPage.getImagesDisplayed()) {
+            softAssert.assertTrue(i.isDisplayed());
+        }
+    }
+
+    @Step("Assert 4 texts on the Index Page under icons")
+    public void assertTextUnderImages() {
+        indexPage = new IndexPage(driver);
+        assertThat(indexPage.getTextUnderImages()).as("Wrong actual text under icons").isEqualTo(expectedText);
+    }
+
+    @Step("Assert iframe with “Frame Button” exist")
+    public void assertIframe() {
+        indexPage = new IndexPage(driver);
+        softAssert.assertTrue(indexPage.getFrameElement().isDisplayed());
+        softAssert.assertAll();
+    }
+
+    @Step("Frame Button exists, Switch to the iframe and check that there is “Frame Button")
+    public void assertButtonInFrame() {
+        indexPage = new IndexPage(driver);
+        driver.switchTo().frame(iFrameElement);
+        assertThat(indexPage.getFrameButton().isDisplayed());
+    }
+
+    @Step("5 items in the left section")
+    public void assertLeftItemsText() {
+        leftSection = new LeftSection(driver);
+        assertThat(leftSection.getNavigationMenuText()).as("Incorrect items in the left section").isEqualTo(names);
+    }
+
+    @Step("Different elements page is opened with proper title")
+    public void assertDiffElementsTitle() {
+        //header = new Header(driver);
+        assertThat(driver.getTitle()).as("Incorrect diff elements title is displayed").isEqualTo(DIFF_ELEMENTS_TITLE);
+    }
+
+    @Step("Clicking checkboxes and assert checkboxes are ticked")
+    public void assertCheckboxes() {
+        diffElementsPage = new DiffElementsPage(driver);
+        softAssert.assertTrue(diffElementsPage.selectWaterCheckbox().isSelected());
+        softAssert.assertTrue(diffElementsPage.selectWindCheckbox().isSelected());
+    }
+
+    @Step("Selecting selen radiobutton")
+    public void assertRadioButton() {
+        diffElementsPage = new DiffElementsPage(driver);
+        softAssert.assertTrue(diffElementsPage.selectSelenRadioButton().isSelected());
+        softAssert.assertAll();
+    }
+
+    @Step("Select yellow dropdown")
+    public void assertDropdown() {
+        diffElementsPage = new DiffElementsPage(driver);
+        softAssert.assertTrue(diffElementsPage.selectYellowDropdown().isSelected());
+        softAssert.assertAll();
+    }
+
+    @Step("Check logpanel text")
+    public void assertLogpanel() {
+        diffElementsPage = new DiffElementsPage(driver);
+        assertThat(diffElementsPage.getWaterCheckboxLog().isDisplayed());
+        assertThat(diffElementsPage.getWindCheckboxLog().isDisplayed());
+        assertThat(diffElementsPage.getMetalRadiobuttonLog().isDisplayed());
+        assertThat(diffElementsPage.getColorsDropdownLog().isDisplayed());
+        softAssert.assertAll();
```

Comment:
If softAssert is not used for verifications in a method, there is no purpose to use assertAll()